### PR TITLE
Fix/1370 efficientad validation and pretraining images

### PR DESCRIPTION
**Pull request by jpcbertoldo**
## Description

Fixes issues A and C in https://github.com/openvinotoolkit/anomalib/issues/1370

A: make it possible to split the train set to make a validation set.

C: make it possible to point to a directory with the images that should be used to the penalty term in the student loss of efficientad.

## Changes

<details>
<summary>Describe the changes you made</summary>

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor (non-breaking change which refactors the code base)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

File: src/anomalib/models/efficient_ad/lightning_model.py [hunk 1]
```diff
@@ -63,6 +63,9 @@ class EfficientAd(AnomalyModule):
         pad_maps (bool): relevant if padding is set to False. In this case, pad_maps = True pads the
             output anomaly maps so that their size matches the size in the padding = True case.
         batch_size (int): batch size for imagenet dataloader
+        pretraining_images_dir (str): path to folder with images used to pretrain the teacher model
+            TODO note in PR: the vocabulary is not consistent with the paper, where they call it "pretraining dataset"
```
Comment (jpcbertoldo) [inline src/anomalib/models/efficient_ad/lightning_model.py:67]:
```suggestion
```

**Comment by jpcbertoldo** [comment]
I additionally adressed an issue raise in this discussion https://#discussioncomment-7127646

The option `pad_maps` was removed to make the interface simpler. The behavior adopted is equivalent to `padding XOR pad_maps` (either the convolutions or the inference score maps are padded).

File: src/anomalib/models/efficient_ad/lightning_model.py [hunk 2]
```diff
@@ -59,10 +59,10 @@ class EfficientAd(AnomalyModule):
         model_size (str): size of student and teacher model
         lr (float): learning rate
         weight_decay (float): optimizer weight decay
-        padding (bool): use padding in convoluional layers
-        pad_maps (bool): relevant if padding is set to False. In this case, pad_maps = True pads the
-            output anomaly maps so that their size matches the size in the padding = True case.
+        padding (bool): use padding in convoluional layers of the student/teacher architecture
```
Comment (Seanny123) [inline src/anomalib/models/efficient_ad/lightning_model.py:62]: spelling/grammar nitpick
```suggestion
        padding (bool): use padding in the convolutional layers of the student/teacher architecture
```

File: src/anomalib/models/efficient_ad/lightning_model.py [hunk 3]
```diff
@@ -73,8 +73,8 @@ def __init__(
         lr: float = 0.0001,
         weight_decay: float = 0.00001,
         padding: bool = False,
-        pad_maps: bool = True,
         batch_size: int = 1,
+        pretraining_images_dir: str = "./datasets/imagenette",
```
Comment (Seanny123) [inline src/anomalib/models/efficient_ad/lightning_model.py:77]: Would it make sense to make this parameter optional? That way, if it is `None` you can still do the automatic download of the Imagenet dataset.
